resources/agent.py

```diff
@@ -141,7 +144,18 @@ def post(self, id):
         if not agent:
             return {"message": "Agent not found"}, 404
 
+        text_splitter = MistralTokenizer.v3(is_tekken=True)
+
         query = request.json.get("query")
+
+        tokens_question = text_splitter.encode_chat_completion(ChatCompletionRequest(messages=[UserMessage(content=query)])).tokens
+
+        # Tokenizer doesn't like including the first two tokens when
+        # decoding...
+        if len(tokens_question) > MAX_TOKENS_QUESTION+2:
+            log.debug("Question too big, truncating...")
```

Comment:
I'm not sure if we ever want to truncate the asked question. Or at least hopefully, we never would need to unless the user asked a ridiculously long question. I think we should only look at the total of current question+message history and start to drop `previous_messages` if the number of tokens becomes too large. I think all the token counting and truncation could happen within the `ask` function, let me know if this idea is off base.